controller/main.py

```diff
@@ -0,0 +1,161 @@
+"""
+Module to analyze and explain PowerPoint presentations using OpenAI's GPT-3.5.
+
+This module is designed as a command-line interface (CLI) application. It accepts a path to a PowerPoint
+presentation and outputs a JSON file with explanations for each slide. It uses OpenAI's GPT-3.5 model to generate the
+explanations.
+
+Usage:
+    python main.py "path/to/presentation.pptx" --log DEBUG --output my_output --dir "path/to/output/dir"
+"""
+
+import logging
+from pptx_parser_ import parser
+from prompt_generator.generate_prompt import PromptGenerator
+import asyncio
+import openai
+import os
+import json
+from dotenv import load_dotenv
+import sys
+from datetime import datetime
+import argparse
+import backoff
+import time
+
+# Configure logging
+logging.basicConfig(filename='program_logs.log', level=logging.INFO, format='[%(levelname)s] %(message)s')
+# Load environment variables, including the API key
+load_dotenv()
+api_key = os.getenv("OPENAI_API_KEY")
+
+
+@backoff.on_exception(backoff.expo, openai.error.RateLimitError)
+async def send_prompt(prompt: str, slide_number: int) -> str:
+    """
+       Send a prompt to GPT-3 and return its response. Function decorated with backoff package to deal with exceptions:
+       if an exceptions is raised, the prompt is sent again until there is no exception raised or 30 seconds
+       passed from the first try
+
+       Args:
+           prompt (str): The prompt to send to GPT-3.
+           slide_number (int): The number of the current slide.
+
+       Returns:
+           str: The response from GPT-3.
+       """
+
+    try:
+        openai.api_key = api_key
+        logging.info("Sending prompt...")
+
+        response = await openai.ChatCompletion.acreate(
+            model="gpt-3.5-turbo",
+            messages=[{"role": "user", "content": prompt}])
+        logging.info(f"response from gpt to slide {slide_number}: {response.choices[0].message.content}")
+
+        return f"Slide {slide_number}: {response.choices[0].message.content}"
+
+    except openai.error.RateLimitError as e:
+        logging.error(f"Rate limit error on slide {slide_number}. Waiting for 60 seconds before retrying.")
```

Comment:
it is a long time to wait, think about the customer who's waiting for your code.